packages/location-management/src/components/LocationUnitAddEdit/index.tsx

```diff
@@ -0,0 +1,169 @@
+    hierarchy.push(data);
+  }
+
+  return hierarchy;
+}
+
+export const LocationUnitAddEdit: React.FC = () => {
+  const params: { id: string } = useParams();
+  const accessToken = useSelector((state) => getAccessToken(state) as string);
+  const [locationUnitGroup, setLocationUnitGroup] = useState<LocationUnitGroup[]>([]);
+  const [LocationUnitDetail, setLocationUnitDetail] = useState<FormField | undefined>(undefined);
+  const Treedata = useSelector(
+    (state) => (getAllHierarchiesArray(state) as unknown) as ParsedHierarchyNode[]
+  );
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (params.id) {
+      const serve = new OpenSRPService(
+        accessToken,
+        API_BASE_URL,
+        `location/${params.id}?is_jurisdiction=true`
+      );
+      serve
+        .list()
+        .then((response: LocationUnit) => {
+          setLocationUnitDetail({
+            name: response.properties.name,
+            parentId: response.properties.parentId,
+            status: response.properties.status,
+            externalId: response.properties.externalId,
+            locationTags: response.locationTags?.map((loc) => loc.id),
+            geometry: JSON.stringify(response.geometry),
+            type: response.type,
+          });
+        })
+        .catch(() => sendErrorNotification('An error occurred'));
+    }
+  }, [accessToken, params.id]);
+
+  useEffect(() => {
+    if (!locationUnitGroup.length) {
+      const serve = new OpenSRPService(accessToken, API_BASE_URL, LOCATION_UNIT_GROUP_ALL);
+      serve
+        .list()
+        .then((response: LocationUnitGroup[]) => {
+          setLocationUnitGroup(response);
+        })
+        .catch(() => sendErrorNotification('An error occurred'));
+    }
+  }, [accessToken, locationUnitGroup.length]);
+
+  useEffect(() => {
+    if (!Treedata.length) {
+      getBaseTreeNode(accessToken)
+        .then((response) => {
+          dispatch(fetchLocationUnits(response));
+          getHierarchy(response, accessToken)
+            .then((hierarchy) => {
+              hierarchy.forEach((hier) => {
+                const processed = generateJurisdictionTree(hier);
+                dispatch(fetchAllHierarchies(processed.model));
+              });
+            })
+            .catch(() => sendErrorNotification('An error occurred'));
+        })
+        .catch(() => sendErrorNotification('An error occurred'));
+    }
+  }, [Treedata, accessToken, dispatch]);
+
+  if (!locationUnitGroup.length || !Treedata.length || (params.id && !LocationUnitDetail))
+    return (
+      <Spin
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+          minHeight: '85vh',
+        }}
+        size={'large'}
+      />
+    );
+
+  return (
+    <Row className="layout-content">
+      <Helmet>
+        <title>{params.id ? 'Edit' : 'Add'} Location Unit Group</title>
```

Comment:
Let's move  Location Unit Group to lang file

Reply:
@dubdabasoduba 🤔💭 its just my idea , shouldn't it be "Editing XYZTeamName" when editing a location unit or location unit group... and also the same approaches for rest of the components and places ?

Reply:
Sure. I have raised an issue to discuss with @rowo

Reply:
done @kahummer